lambdaclass/mirra_backend
# [GH-782] username changes

## Motivation

The client needed some changes in the users.username behavior :
- usename should be able to be repeated 
- we need the same length validation that the client has

Closes #782

## Summary of changes

- Drop unique index game_id, username in the users table
- Add length validation to the user changeset

## How to test it?

- Create 2 users in the repo backend using:
```
{:ok, user1} = GameBackend.Users.create_guest_user
{:ok, user2} = GameBackend.Users.create_guest_user
```

- Then try to create an invalid name:
  - Short name:
    - `curl -X PUT -d username="a"  http://localhost:4001/users/user1.id`
  - Long name:
    - `curl -X PUT -d username="agustinesco"  http://localhost:4001/users/user1.id`
  - Repeated name
     - `curl -X PUT -d username="agustin"  http://localhost:4001/users/user1.id`
     - `curl -X PUT -d username="agustin"  http://localhost:4001/users/user2.id`

Testing champions users:
- Creating user
  - `{:ok, champions_user} = Champions.Users.register("something")`
- Try to change the name to anything
  - `curl -X PUT -d username="ag"  http://localhost:4001/users/champions_user.id`
It should change without any restriction


## Checklist
